src/com/tms/lesson6/Searchable.java

```diff
@@ -0,0 +1,15 @@
+package com.tms.lesson6;
+
+import com.tms.lesson5.Director;
+import com.tms.lesson5.Employee;
+
+public interface Searchable {
```

Comment:
no need to use both methods: getEmployeeByName  and getEmployeeByNameDeep - only one.
Implementation will use different realisations.
SimpleSearch implements Searchable
DeepSearch implements Searchable

Reply:
fixed